components/apimgt/org.wso2.carbon.apimgt.api/src/main/java/org/wso2/carbon/apimgt/api/APIProvider.java

```diff
@@ -1598,6 +1618,36 @@ List<OperationPolicyData> getAllAPISpecificOperationPolicies(String apiUUID, Str
     void deleteOperationPolicyById(String policyId, String organization) throws APIManagementException;
 
     /**
+     * Delete an API endpoint by providing the endpoint ID.
+     *
+     * @param endpointId     API Endpoint UUID
+     * @throws APIManagementException
+     */
+    void deleteAPIEndpointById(String endpointId) throws APIManagementException;
+
+    /**
+     *  Update an endpoint by providing the endpoint ID.
+     *
+     * @param endpointUUID Unique identifier of endpoint
+     * @param apiEndpoint  Endpoint with updated details
+     * @return
+     * @throws APIManagementException
+     */
+    APIEndpointInfo updateAPIEndpoint(String endpointUUID, APIEndpointInfo apiEndpoint)
```

Comment:
Since endpointUUID is defined inside APIEndpointInfo model itself, can we set the uuid there and not pass `endpointUUID` explicitly here?

Reply:
yes, correct after the API provider layer we do not explicitly want the UUID of an EP.

Reply:
can you fix this